### ISSUE TITLE
howto to use sockets

### ISSUE THREAD

**Issue by Licenser** [satom99/phx_raws#2]
Hi first of all thanks for creating this, I was quite shocked to see that WebSockets are not supported in Phoenix and relieved to find this :D

That said for those of us less versed in Phoenix it would be really awesome to add a line or two how to add this to the routes.

**Comment by Licenser** [comment]
I should follow up here the info I was missing here was:

```elixir
defmodule Example.Endpoint do
  use Phoenix.Endpoint, otp_app: :example
  socket "/websockets", Example.Gateway

  # ...
end
```

I can make a PR to include that in the readme but it might be simpler just to copy&paste that snippet :)

**Comment by satom99** [comment]
I agree a better example would help a lot, this is something I will tackle as I'm done with #1. 🙂 

For the record and in the meantime, I would like to reference a [topic](https://elixirforum.com/t/plan-old-websockets-in-phoenix-but-without-magic/10074/18) you created on the forum that might be helpful for other devs to read - specifically the marked answer as it contains a nice example.

**Comment by dcy** [comment]
HI! which url is right? @Licenser

**Comment by Licenser** [comment]
huh?

**Comment by dcy** [comment]
```new Websocket("url")```
```var ws = new WebSocket("ws://localhost:4001/websockets")``` and ```var ws = new WebSocket("ws://localhost:4001/websockets/websocket")``` don't work.

**Comment by satom99** [comment]
@dcy [this](https://github.com/phoenixframework/phoenix/blob/7329665be6ab99a6f5d6379288fba3a6bf54b45c/lib/phoenix/endpoint/cowboy_handler.ex#L93) is what Phoenix sets as your socket path. It essentially joins the string given on the `socket` macro and the atom provided on the `transports` macro.

That-is, if you have `socket "", Example.Gateway` and `transports :gateway, ...` then your path would be an empty string joined with the stringified atom, thus `/gateway`.

**Comment by dcy** [comment]
What's the diffrence between phx_raw and [Custom dispatch options](https://hexdocs.pm/phoenix/Phoenix.Endpoint.CowboyHandler.html#module-custom-dispatch-options)?

**Comment by satom99** [comment]
Essentially *phx_raws* serves as an utility for those who don't want to both have to add an own cowboy handler nor configure the dispatch options every time a socket is added or a new project to be started.

Besides, this follows Phoenix' semantics and uses its code - thus no functionality is re-implemented unnecessarily.

**Comment by Licenser** [comment]
@dcy sorry I don't remember, I ended up using the custom dispatch as even with phx_raws phoenix kept messing with my code so I couldn't use it. (not phx_raw's fault!)

**Comment by bernardo-martinez** [comment]
> @dcy sorry I don't remember, I ended up using the custom dispatch as even with phx_raws phoenix kept messing with my code so I couldn't use it. (not phx_raw's fault!)

how did you do the dispatch to be compatible with existing Router/Endpoint in your project? can you share an example?
